pkg/trace/writer/stats.go

```diff
@@ -168,112 +158,143 @@ func (w *StatsWriter) sendPayloads() {
 	w.payloads = w.payloads[:0]
 }
 
-// buildPayloads returns a set of payload to send out, each paylods guaranteed
-// to have the number of stats buckets under the given maximum.
-func (w *StatsWriter) buildPayloads(s []stats.Bucket, maxEntriesPerPayloads int) ([]*stats.Payload, int, int) {
-	if len(s) == 0 {
-		return []*stats.Payload{}, 0, 0
+// encodePayload encodes the payload as Gzipped msgPack into w.
+func encodePayload(w io.Writer, payload pb.StatsPayload) error {
+	gz, err := gzip.NewWriterLevel(w, gzip.BestSpeed)
+	if err != nil {
+		return err
+	}
+	_, err = gz.Write([]byte{v1})
```

Comment:
This is probably too late to change that, but it's an odd way of versioning a API. Usually we version the api path. If I understand correctly the payload format is no longer valid msgpack since there is an extra byte.

Reply:
I removed the version bit.